SS2-Project/Assets/Starstorm2/Modules/SS2EventCard.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using UnityEngine;
+using Moonstorm.AddressableAssets;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using RoR2;
+using R2API;
+using UnityEngine.AddressableAssets;
+
+namespace Moonstorm.Starstorm2
+{
+    public class SS2EventCard : EventCard
+    {
+        [Serializable]
+        public class EventVFX
+        {
+            public R2API.DirectorAPI.Stage stageEnum;
+            public string customStageName;
+            public GameObject effectPrefab;
+
+            public GameObject EffectPrefab { get; internal set; }
+        }
+
+        private static List<SS2EventCard> instances = new List<SS2EventCard>();
+        public AddressableItemDef requiredItem;
+        public GameObject fallbackVFX;
+        public EventVFX[] eventVFX = Array.Empty<EventVFX>();
+        
+        public ReadOnlyDictionary<R2API.DirectorAPI.Stage, GameObject> vanillaStageToFXPrefab;
+        public ReadOnlyDictionary<string, GameObject> customStageToFXPrefab;
+
+        private void Awake()
+        {
+            instances?.Add(this);    
+        }
+        private void OnDestroy()
+        {
+            instances?.Remove(this);
+        }
+
+        [SystemInitializer]
+        private static void SystemInit()
+        {
+            EventCatalog.resourceAvailability.CallWhenAvailable(() =>
```

Comment:
Due to using MSU's AddressableAssets in the form of the AddressableItemDef, this should use AddressableAssets.OnAddressableAssetsLoaded to ensure the itemdef is populated